### PR TITLE
RPD immitation TGUI

### PR DESCRIPTION
**Pull request by CrazyAmphibian**
[ui]
## What this does
changes the RPD's UI to resemble stuff made with TGUI (it's just some HTML and CSS changes.)
Capture
Capture2

## Why it's good
for one, visual consistency with other UI. also it looks better than a plain HTML listing. it also adds some more functionality by highlighting what is selected so you don't have to scroll up to check. (it also fixes a bug with the favorites list movement (SO UNATOMIC!!!11!)

## Changelog
:cl:
 * tweak: the RPD now has a different look to it that is in line with other machines.
 * bugfix: the RPD favorites menu will now show the correct movement arrows

### CODE REVIEW

**Comment by CrazyAmphibian** [comment]
i changed the selected schematic color and favorites blurb to white to make them more noticable
Capture

**Comment by SonixApache** [comment]
jesus that's ugly

**Comment by CrazyAmphibian** [comment]
> jesus that's ugly

thank you for your very constructive and insightful feedback.

**Comment by SonixApache** [comment]
ok
For one, it follows zero standards that we actually employ - the font is different, the font color is different, the background color is different, it's all non-standard. And the beige looks fucking ugly.
That aside, the layer selector is now next to the options, rather than off-set where it's supposed to be, which means that it's going to move constantly - that's horrible.

**Comment by CrazyAmphibian** [comment]
i took the background color from nanotrassen.css (@code\modules\html_interface\nanotrasen\nanotrassen.css).
i can change the font to what that uses, too.
the text color i took from what #35326 used for its non-label text
i'll see what i can do about making the layer selector fixed position again.

**Comment by CrazyAmphibian** [comment]
Capture
Capture2

now has a fixed x position. it also uses the same exact font as other stuff now.

**Comment by CrazyAmphibian** [comment]
Capture
pipe layers now have a gradient over them to make them less flat in appearance (and like a pipe to how it was before)

**Comment by falcon2346** [comment]
>the font color is different,  the background color is different

Today I learned that Sonix has never looked at an APC, Air Alarm, or the CAC, which all use the same background color and font color, though two of them also have a decal.

**Comment by SonixApache** [comment]
@falcon2346 Here's, for comparison, the APC, which uses nanoUI. It looks different.
image vs image
image  image
And here's the merch computer and GPS, which as far as I know use TGUI. They also look different.
Maybe it's the size of the font, maybe it's the contrast between the text color and background; either way, it looks bad.

File: code/modules/RCD/RCD.dm [hunk 1]
```diff
@@ -86,9 +86,9 @@
 		dropped_by.hud_used.toggle_show_schematics_display(null,1, src)
 
 /obj/item/device/rcd/attack_self(var/mob/user)
-	var/turf/T = get_turf(src)
-	if(T?.z != z_last_checked)
-		rebuild_ui()
+	//var/turf/T = get_turf(src)
+	//if(T?.z != z_last_checked) i don't know why this in in here. commented out instead of removed in case it FSU
```
Comment (west3436) [inline code/modules/RCD/RCD.dm:90]: The commented-out items are for multi-z purposes: unless they harm the performance of your UI they should probably stay in uncommented.

Comment (CrazyAmphibian) [reply code/modules/RCD/RCD.dm:90]: that's not what that's for. looking at the varible it usues it is only used in 2 places, once when it is defined upon item creation, and the second there when it is read. 
it's incredibly jank shitcode that i have no idea the function of and removal seems to affect nothing.
i needed to remove that check to allow redrawing of the UI to display which pipe is currently being selected

Comment (west3436) [reply code/modules/RCD/RCD.dm:90]: \>git blame
\>see who wrote it
lmao disregard remove at will

**Comment by falcon2346** [comment]
> @falcon2346 Here's, for comparison, the APC, which uses nanoUI. It looks different. image vs image ![image](https://private-user-images.githubusercontent.com/8468269/305129483-7957c55f-e2d8-463d-92bc-acae02d2b26c.png?

I'm sure you already know this since you cropped the APC image, but the the background color you posted is due to a decal on it with a gradient that transitions to the same color as the one used in the fake TGUI rpd. I simply think it's remarkably unhelpful to leave a "wow this is terrible" and then when asked to elaborate say that it doesn't follow 3 standards, 2 of which you're just incorrect on. 

The font type and sizes should probably be changed to match the one used in real TGUI menus though for sure, consistency is important, Maybe it would be best to add that decal as well.

**Comment by cloutcode** [comment]
oh i wanna fuck you so bad

**Comment by hacker-on-steroids** [comment]
i for one don't particularly appreciate forcing muh darkmode on every UI
thought i'd get used to the RND one but i didn't

**Comment by CrazyAmphibian** [comment]
> i for one don't particularly appreciate forcing muh darkmode on every UI

i think it'd be great to have an option that allows players to change their UI themes, but i don't think that it's possible, and if it is, it's not in the scope of this PR

**Comment by CrazyAmphibian** [comment]
> Maybe it would be best to add that decal as well.

the menu is too tall for the background decal to look sensible on it

**Comment by CrazyAmphibian** [comment]
Capture
i've imported more css (as well as added some more padding) from nanotrassen.css to make it upholant to the standards.
give feedback on this @SonixApache  (for some reason images weren't being sent on the local server idk why)

**Comment by jellyveggie2** [comment]
Copy-pasting bits and pieces of nanotrassen.css onto RCD.css seems to me like a bad idea. You're risking missing some important piece, and you're also depending on future coders to remember they have to update RCD.css whenever they update nanotrasen.css (they won't).

You should just append nanotrasen.css to the <head> using a <link> tag so it loads the file itself, and trim RCD.css down to whatever you actually want to be different. Use `new/datum/html_interface/()`'s `head` param to do so. 
You might even want to switch out `var/datum/html_interface/rcd/interface` for `var/datum/html_interface/nanotrasen/interface` and append RCD.css instead, so conceptually nanotrasen.css acts as the base and RCD.css is what gets added on top? Though ironically, that would have nanotrasen.css load _after_ RCD.css and thus take precedence over any rules they may have in common, which might be a hindrance.

**Comment by CrazyAmphibian** [comment]
> you're also depending on future coders to remember they have to update RCD.css whenever they update nanotrasen.css (they won't).

I figured that, since the RPD menu is its own bit of code, it'd be fine to isolate it, but i suppose that's a fair point if the UI theme ever gets chaged

> that would have nanotrasen.css load _after_ RCD.css and thus take precedence over any rules they may have in common, which might be a hindrance.

i don't think i got your explanation of why this would be, but HTML files can load any number of CSS files, and any conflicts are overwritten with the last CSS loaded in taking priority over the first. If i did a deep dive into CSS loading with this i'm sure i could get them loading in the right order

**Comment by jellyveggie2** [comment]
> i don't think i got your explanation of why this would be, but HTML files can load any number of CSS files, and any conflicts are overwritten with the last CSS loaded in taking priority over the first. If i did a deep dive into CSS loading with this i'm sure i could get them loading in the right order

Just an issue with how `/datum/html_interface/nanotrasen/New()` is implemented:
```dm
src.head = src.head + "<link rel=\"stylesheet\" type=\"text/css\" href=\"nanotrasen.css\" />"
``` 

This would put nanotrasen.css last on head and (as far as I understand) cause it to load last. If there's a way to force a different loading order without changing their actual order in <head> that works in IE and doesn't involve a load of !important, that would be nice.

Otherwise, a fix could be
```dm
src.head = "<link rel=\"stylesheet\" type=\"text/css\" href=\"nanotrasen.css\" />" + src.head
```
But I wouldn't be surprised if reversing that somehow revealed a number of issues across all other UIs using `/datum/html_interface/nanotrasen/New()`.

**Comment by CrazyAmphibian** [comment]
the RCD doesn't use `/datum/html_interface/nanotrasen/New()`, rather  `/datum/html_interface/rcd/New()` \@code/modules/html_interface/RCD/RCD.dm
the thing for me to figure out is how the CSS is sent to the client so i can modify the head

**Comment by jellyveggie2** [comment]
> the RCD doesn't use `/datum/html_interface/nanotrasen/New()`, rather `/datum/html_interface/rcd/New()`

I... yes, that's what I was talking about at first. About how I felt switching out `/datum/html_interface/rcd/` it's currently using and using `/datum/html_interface/nanotrasen/` as the base made more sense conceptually (to me, at least) but might pose a small issue on a technical level.
Sorry if I wasn't clear.

**Comment by CrazyAmphibian** [comment]
> using `/datum/html_interface/nanotrasen/` as the base...but might pose a small issue on a technical level.

well, there's code in the rcd.dm, such as sending pipe icons that wouldn't be in the nanotrassen.dm, so it'd probably be a lot easier to just send over nanotrassen stuff in the rcd code instead

**Comment by CrazyAmphibian** [comment]
gentlemen. (images still are borked on my local server)
Capture
Capture2

**Comment by SonixApache** [comment]
looks fine to me
as long as the layer selector doesn't ever move i think this is a good addition :+1:

**Comment by CrazyAmphibian** [comment]
> as long as the layer selector doesn't ever move i think this is a good addition 👍

it'll always remain right above the frequency/ID options with a fixed distance from the left side of the menu

**Comment by Kurfursten** [comment]
Change bad! Blank white soulful.

**Comment by Servalfang** [comment]
why did you merge this

**Comment by Wouju** [comment]
sovl is gone, it's over

**Comment by Kurfursten** [comment]
Had enough support to warrant trying (like the R&D one too)

Personally I don't like either but you are welcome to click the revert button and see if you can get support

**Comment by CrazyAmphibian** [comment]
a full revert wouldn't be the best choice since this did fix a bug and added functionality the old UI didn't have (eg, fixing favorites arrows being wrong, and showing you which schematic was selected without having to scroll up), so a revert would be an objective downgrade in terms of function.

**Comment by jwhitak** [comment]
> a full revert wouldn't be the best choice since this did fix a bug and added functionality the old UI didn't have (eg, fixing favorites arrows being wrong, and showing you which schematic was selected without having to scroll up), so a revert would be an objective downgrade in terms of function.

And this is why PRs should be atomic. If they get reverted because of some bad thing, all the other things are lost as well. Even if those other things were good. I say that as I did a crazy unatomic cargo computer UI change kinda recently...

**Pull request merged**